src/views/credentials/addCredentialModal.tsx

```diff
@@ -52,283 +86,308 @@ const useCredentialForm = (credentialType: string | undefined, credential?: Cred
-          <FormHelperText>
-            Please paste your private SSH key here. This key will be used to authenticate your access to the system.
-          </FormHelperText>
-        </FormGroup>
-        <FormGroup label="SSH passphrase" fieldId="ssh_passphrase">
-          <TextInput
-            value={formData?.ssh_passphrase}
-            placeholder="Enter SSH passphrase (optional)"
-            type="password"
-            id="ssh_passphrase"
-            name="ssh_passphrase"
-            onChange={event => handleInputChange('ssh_passphrase', (event.target as HTMLInputElement).value)}
-            ouiaId="ssh_passphrase"
-          />
-        </FormGroup>
-      </React.Fragment>
-    )}
 
-    {/* Render "Become" fields only if network is selected and authType is Username/Password or SSH Key */}
-    {typeValue === 'network' && (
-      <React.Fragment>
-        <FormGroup label="Become Method" fieldId="become_method">
+      {/* Render Authentication Type dropdown only if needed based on the credential type */}
+      {(typeValue === 'network' || typeValue === 'openshift') && (
+        <FormGroup label="Authentication Type" fieldId="auth_type">
           <SimpleDropdown
-            label={formData?.become_method || 'Select option'}
-            menuToggleOuiaId="become_method"
+            label={authType}
+            menuToggleOuiaId="auth_type"
             variant="default"
             isFullWidth
-            onSelect={item => handleInputChange('become_method', (item !== 'Select option' && item) || '')}
-            dropdownItems={[
-              { item: 'Select option', ouiaId: 'default' },
-              { item: 'sudo', ouiaId: 'sudo' },
-              { item: 'su', ouiaId: 'su' },
-              { item: 'pbrun', ouiaId: 'pbrun' },
-              { item: 'pfexec', ouiaId: 'pfexec' },
-              { item: 'doas', ouiaId: 'doas' },
-              { item: 'dzdo', ouiaId: 'dzdo' },
-              { item: 'ksu', ouiaId: 'ksu' },
-              { item: 'runas', ouiaId: 'runas' }
-            ]}
+            onSelect={item => setAuthType(item)}
+            dropdownItems={
+              (typeValue === 'network' && [
+                { item: 'Username and Password', ouiaId: 'password' },
+                { item: 'SSH Key', ouiaId: 'ssh_key' }
+              ]) || [
+                { item: 'Token', ouiaId: 'auth_token' },
+                { item: 'Username and Password', ouiaId: 'password' }
+              ]
+            }
           />
         </FormGroup>
-        <FormGroup label="Become User" fieldId="become_user">
+      )}
+
+      {/* Conditional rendering for Token input */}
+      {authType === 'Token' && (
+        <FormGroup label="Token" isRequired fieldId="auth_token">
           <TextInput
-            value={formData?.become_user}
-            placeholder="Enter become user (optional)"
+            value={formData?.auth_token}
+            placeholder={maskedFields.auth_token ? '****' : 'Enter Token'}
```

Comment:
highly recommended 6:
and finally, expand this to the other "masked fields"

let's convert all `maskedFields.[SOME FIELD]` into `credential?.has_auth_token`

optional:
follow the ternary pattern we've been using elsewhere
```
(credential?.has_auth_token && '****') || 'Enter Token'

```